STUDIO/org.openl.rules.webstudio/src/org/openl/rules/rest/DesignTimeRepositoryController.java

```diff
@@ -202,7 +215,20 @@ public ProjectViewModel createProjectFromZip(@DesignRepository("repo-name") Repo
             @Parameter(description = "repos.create-project-from-zip.param.template.desc", content = @Content(encoding = @Encoding(contentType = "application/zip"))) @RequestParam("template") MultipartFile file,
             @Parameter(description = "repos.create-project-from-zip.param.overwrite.desc") @RequestParam(value = "overwrite", required = false, defaultValue = "false") Boolean overwrite) throws IOException,
                                                                                                                                                                                            JAXBException {
-        SecurityChecker.allow(overwrite ? Privileges.EDIT_PROJECTS : Privileges.CREATE_PROJECTS);
+        if (overwrite) {
+            if (!designRepositoryAclService
+                .isGranted(repository.getId(),
+                    repository.supports().mappedFolders() && !StringUtils.isBlank(path) ? RepositoryAclServiceImpl

```

Comment:
Extract arguments to the variables. It is very hard to read and debug such constructions.